### PR TITLE
Add host network

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #909

All the description is in the issue mentioned above.

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
Running verification here:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/1561/console

**Comment by petr-balogh** [comment]
Just applied WA when deployment was running: 
```
oc rsh -n openshift-storage rook-ceph-tools-56db68cc99-bvnls
sh-4.2# for i in $(ceph osd lspools |awk '{print$2}'); do ceph osd pool set $i crush_rule replicated_rule; done
set pool 1 crush_rule to replicated_rule
set pool 2 crush_rule to replicated_rule
set pool 3 crush_rule to replicated_rule
set pool 4 crush_rule to replicated_rule
set pool 5 crush_rule to replicated_rule
set pool 6 crush_rule to replicated_rule
set pool 7 crush_rule to replicated_rule
set pool 8 crush_rule to replicated_rule
set pool 9 crush_rule to replicated_rule
set pool 10 crush_rule to replicated_rule
```
So we should see the deployment passed I hope.

**Comment by petr-balogh** [comment]
```
oc get pod -n openshift-storage
NAME                                                         READY   STATUS      RESTARTS   AGE
csi-cephfsplugin-8sx9t                                       3/3     Running     0          7m11s
csi-cephfsplugin-hxr2d                                       3/3     Running     0          7m11s
csi-cephfsplugin-provisioner-57f65684f4-2558x                4/4     Running     0          7m11s
csi-cephfsplugin-provisioner-57f65684f4-ch2d8                4/4     Running     0          7m11s
csi-cephfsplugin-qfzqw                                       3/3     Running     0          7m11s
csi-rbdplugin-c9wgj                                          3/3     Running     0          7m11s
csi-rbdplugin-cfwn6                                          3/3     Running     0          7m11s
csi-rbdplugin-mml6m                                          3/3     Running     0          7m11s
csi-rbdplugin-provisioner-54985c744b-bmr8f                   5/5     Running     0          7m11s
csi-rbdplugin-provisioner-54985c744b-g958t                   5/5     Running     0          7m11s
local-storage-operator-bcfd5765f-jswcv                       1/1     Running     0          8m
noobaa-core-0                                                2/2     Running     0          7m10s
noobaa-operator-7c55776bf9-r5sjr                             1/1     Running     0          7m44s
ocs-operator-6cf498ff89-kwp87                                1/1     Running     0          7m44s
rook-ceph-drain-canary-ip-10-0-141-36-f84897496-mxvrt        1/1     Running     0          3m38s
rook-ceph-drain-canary-ip-10-0-147-232-9d87477bb-2g2l7       1/1     Running     0          3m46s
rook-ceph-drain-canary-ip-10-0-171-68-5fb98bd58b-g46ml       1/1     Running     0          3m46s
rook-ceph-mds-rook-ceph-cephfilesystem-a-59476cc94b-ntffc    1/1     Running     0          3m26s
rook-ceph-mds-rook-ceph-cephfilesystem-b-7dd5855dc8-ksv7q    1/1     Running     0          3m25s
rook-ceph-mgr-a-c8c489fc-q4hkg                               1/1     Running     0          4m33s
rook-ceph-mon-a-869d74b564-s699j                             1/1     Running     0          5m51s
rook-ceph-mon-b-5b8f664dc5-tmkdk                             1/1     Running     0          5m21s
rook-ceph-mon-c-769d858c54-j2xzn                             1/1     Running     0          4m51s
rook-ceph-operator-8444cfdc4c-hqwbg                          1/1     Running     0          7m44s
rook-ceph-osd-0-865697b7fc-pkgnr                             1/1     Running     0          3m46s
rook-ceph-osd-1-7568dfbdd8-w8g8f                             1/1     Running     0          3m46s
rook-ceph-osd-2-747596d97f-kkds9                             1/1     Running     0          3m39s
rook-ceph-osd-prepare-example-deviceset-0-0-hxcr5-55lcq      0/1     Completed   0          4m21s
rook-ceph-osd-prepare-example-deviceset-1-0-6g5hc-frlwr      0/1     Completed   0          4m21s
rook-ceph-osd-prepare-example-deviceset-2-0-xwc6x-vdm4q      0/1     Completed   0          4m21s
rook-ceph-rgw-rook-ceph-cephobjectstore-a-6f9fb4f77c-z8jc8   1/1     Running     0          2m14s
rook-ceph-tools-56db68cc99-bvnls                             1/1     Running     0          3m23s
```

File: ocs_ci/deployment/aws.py [hunk 1]
```diff
@@ -77,6 +77,81 @@ def add_volume(self, size=100):
...
+    def host_network_update(self):
+        """
+        Update security group rules for HostNetwork
+        """
+        cluster_id = get_infra_id(self.cluster_path)
+        worker_pattern = f'{cluster_id}-worker*'
+        worker_instances = self.aws.get_instances_by_name_pattern(
+            worker_pattern
+        )
+        security_groups = worker_instances[0]['security_groups']
```
Comment (vasukulkarni) [inline ocs_ci/deployment/aws.py]: Is setting this in first worker sufficient?

Comment (petr-balogh) [reply ocs_ci/deployment/aws.py]: All the workers are using the same security group so I am taking SG just from first one worker.

**Comment by vasukulkarni** [comment]
I think its better to merge and we can use this conf file for testing when needed, the default is not changed anyway and the tier1/workload tests are passing. If there are port changes(which I doubt) we can take it as follow on PR.

**Pull request merged**